### PR TITLE
Grading interface improvements

### LINKED ISSUE #1543: custom mark text should not be allowed to be empty
We have a bug...  if the custom message is blank, the custom point is not displayed in the rubric box (dunno about the hw report).

longer term: Should not be allowed to just fill in a +/- point value for custom, it must also have a message.

### LINKED ISSUE #2023: "no credit" / "full credit" mark should not be editable
"no credit" / "full credit" is based on whether the component is count up or count down.  
It should look visually similar to the other common marks....  but
You should not be able to edit the points or text.  
(It might look a little different to hint that editing is not possible.)
It should always be the first in the list of marks.

### PR DESCRIPTION
**Pull request by jchulton**
Closes #2023 Closes #1543 Takes care of various issues discussed in the pull request WIP Grading interface #2026. Specifically:
Adding the Add Common Mark Button to the box containing non-common marks
Editing Existing mark no longer makes text box get taller
Cancel/Done text now aligns
Clicking to expand the General Comment box improved.

### CODE REVIEW

**Comment by CouleeApps** [comment]
>Editing Existing mark no longer makes text box get taller

Curious why you're doing this. Having vertical scroll in the boxes makes it harder to read them.

File: site/public/js/ta-grading-mark.js [hunk 1]
```diff
@@ -42,18 +42,27 @@ function checkIfSelected(me) {
...
+    if(x==0){
+        color="#e6e6e6";
+        editable="readonly";
+        appearEditable="cursor: not-allowed; border:none; outline:none;";
+    }
+    //onkeyup="autoResizeComment(event) removed from textarea
     return ' \
 <tr id="mark_id-'+num+'-'+x+'" name="mark_'+num+'" class="'+(is_publish ? 'is_publish' : '')+'"'+(is_new ? 'data-newmark="true"' : '')+'> \
-    <td colspan="1" style="'+background+'; width: 90px; text-align: center;"> \
+    <td colspan="1"; width: 90px; text-align: center;"> \
```
Comment (CouleeApps) [inline site/public/js/ta-grading-mark.js]: Should just remove background without removing the opening `style=` attribute.

File: site/public/js/ta-grading-mark.js [hunk 2]
```diff
@@ -42,18 +42,27 @@ function checkIfSelected(me) {
 }
 
 function getMarkView(num, x, is_publish, checked, note, pointValue, precision, min, max, background, gradeable_id, user_id, get_active_version, question_id, your_user_id, is_new) {
+    editable="";
+    appearEditable="";
+    color=background;
```
Comment (CouleeApps) [inline site/public/js/ta-grading-mark.js]: Should use `var` otherwise these will become global variables

File: site/public/js/ta-grading-mark.js [hunk 3]
```diff
@@ -447,16 +454,32 @@ function calculateMarksPoints(question_num) {
...
+            $('#mark_points_custom-' + question_num)[0].value="0";
+        }
+        if (isNaN(custom_points)) {
+            current_points += 0;
+        } 
+        else {
+            if(custom_points!=0){
+                current_points += custom_points;
+                any_selected = true;
+            }
```
Comment (CouleeApps) [inline site/public/js/ta-grading-mark.js]: Selecting only a custom mark with zero points makes it look like you have graded the assignment but shows the score as not graded:

**Comment by CouleeApps** [comment]
Oh I see, https://github.com/Submitty/Submitty/pull/2026#pullrequestreview-124028504 (first bullet point).

Personally I preferred having the boxes expand as it meant you could easily enter multiple lines of text.

**Comment by jchulton** [comment]
Implemented changes above. Please take another look when you can.

**Pull request merged**